### ISSUE TITLE
KotlinVersion is not being taken correctly to the classpath

### ISSUE THREAD

**Issue by klu2** [cloudflightio/autoconfigure-gradle-plugin#88]
If you override the kotlin version like this:

```
autoConfigure {
    kotlin {
        kotlinVersion.set("1.6.21")
    }
}
```

Then you still have the latest kotlin version on the classpath.

If you run the tracker report you see the following:

```
    "compile": [
        {
            "artifact": "org.jetbrains.kotlin:kotlin-bom:1.6.21",
            "classifier": null,
            "type": "bom",
            "trail": [
            ]
        },
        {
            "artifact": "org.jetbrains.kotlin:kotlin-stdlib:1.7.10",
            "classifier": null,
            "type": "jar",
            "trail": [
                "org.jetbrains.kotlin:kotlin-bom:1.6.21",
                "org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10"
            ]
        },
        {
            "artifact": "org.jetbrains.kotlin:kotlin-stdlib-common:1.7.10",
            "classifier": null,
            "type": "jar",
            "trail": [
                "org.jetbrains.kotlin:kotlin-bom:1.6.21",
                "org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10",
                "org.jetbrains.kotlin:kotlin-stdlib:1.7.10"
            ]
        },
        {
            "artifact": "org.jetbrains:annotations:13.0",
            "classifier": null,
            "type": "jar",
            "trail": [
                "org.jetbrains.kotlin:kotlin-bom:1.6.21",
                "org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10",
                "org.jetbrains.kotlin:kotlin-stdlib:1.7.10"
            ]
        },
        {
            "artifact": "org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.7.10",
            "classifier": null,
            "type": "jar",
            "trail": [
                "org.jetbrains.kotlin:kotlin-bom:1.6.21",
                "org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10"
            ]
        },
        {
            "artifact": "org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10",
            "classifier": null,
            "type": "jar",
            "trail": [
            ]
        }
    ]
```

That means, we are applying `org.jetbrains.kotlin:kotlin-bom` with the correct version, but someone (probably the Kotlin plugin) is also adding `org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.7.10`, and that is overriding the version.

The problem seem to be in the `KotlinConfigurePlugin` in that block:

```
        kotlin.sourceSets.maybeCreate("main").dependencies {
            // see https://kotlinlang.org/docs/gradle.html#dependency-on-the-standard-library
            // as we allow clients to override the Kotlin Version, (i.e. to 1.5.20), we also want to ensure
            // that all kotlin standard libraries from exactly that Kotlin version is being added
            // to the dependencies. That's why we add the kotlin-bom in exactly our version here.
            // Without those lines, we would always add the stdlib in the version
            // of the underlying Kotlin Gradle Plugin (1.7.10 at the time of that writing)
            this.implementation(kotlinConfigureExtension.kotlinVersion
                .map { project.dependencies.platform("org.jetbrains.kotlin:kotlin-bom:$it") }
            )
            this.implementation(kotlinConfigureExtension.kotlinVersion
                .map { "org.jetbrains.kotlin:kotlin-stdlib-jdk8" }
            )

```

It seems that the correct `kotlinVersion` is not being taken here

**Comment by klu2** [comment]
it seems that this has to to with the `autoConfigure` plugin, because there is a `KotlinConfigurePluginTest` with a `single-kotlin-module-override-kotlinversion` fixture and that only applies the `io.cloudflight.autoconfigure.kotlin-configure` plugin:

```
kotlinConfigure {
    kotlinVersion = "1.5.20"
}
```

Probably the `AutoConfigureExtension` erronously overrides that.

**Comment by klu2** [comment]
released in version 0.7.0
